### PR TITLE
Cake version upgrade

### PR DESCRIPTION
**Pull request by davidyell**
In reference to #51 

This is an attempt to bump the codebase up to using CakePHP 3.5. It also refactors the combined methods, into a more verbose and descriptive public api.

I have introduced a minimum, but quite lax, requirement for the cake version and the php version. So the plugin can ensure it's environment.

### CODE REVIEW

**Comment by davidyell** [comment]
Do you have any suggestions for the implementation of the Query's `andWhere` and `select` methods, as they are mapped with an `@method` call in the interface now.

I can deal with the PHP Stan and PHP CS, when we're ready to merge 😄

**Comment by ADmad** [comment]
So I guess you didn't heed my advice of making separate PRs per roadmap item, starting with just updating deprecated core methods/props.

**Comment by davidyell** [comment]
No I did. I even asked you on Slack if the deprecated methods should reflect the core, and you said yes.

All this PR does is change the Cake version, fix the deprecated methods and refactor the mixed mode methods. The test suite still passes just fine.

If you'd like me to detail the changes, I'm happy to write them all up to make it easier to review.

**Comment by ADmad** [comment]
> fix the deprecated methods and refactor the mixed mode methods. The test suite still passes just fine.

Updating core deprecated methods and splitting mix mode methods provided by this plugin's classes should have been separate PRs. So should have been changes like changes config "endpoint" to "name" for `Endpoint` class.

> The test suite still passes just fine. 

As reported by codecov there is coverage drop 6.17%. That is because you updated tests for old mix mode methods instead of adding new tests for the new methods. Also few changes you have done are not backwards compatible, I'll comment on them on the diff.

That's fine we can sort it out, just it would have much easier if we were dealing will smaller PRs.

> Also, this isn't a great way to react to someone trying to contribute code to your project.

Nor is it great to ask for how to go about submitting changes and then not following the suggestion given. :slightly_smiling_face: 

To quote myself on slack "One PR per roadmap item. Each PR can have one or more commits as necessary"

File: composer.json [hunk 1]
```diff
@@ -39,9 +39,10 @@
         "irc": "irc://irc.freenode.org/muffin"
     },
     "require": {
+        "php": ">=5.6.0",
+        "cakephp/cakephp": "~3.4"
```
Comment (ADmad) [inline composer.json]: With 3.6 release round the corner we can just jump to that. Later when also changes are done we can see if there are deprecation notices if we try to lower the constraints to 3.5 or 3.4.

So please change the constraint to `3.6.0-beta2` or `dev-3.next as 3.6.0`.

**Comment by davidyell** [comment]
Closing in favour of a more specific set of pull requests.

**Pull request closed without merging**